controllers/util/k8sutil.go

```diff
@@ -23,6 +23,9 @@ const (
 	// this value is empty if the operator is running with clusterScope.
 	WatchNamespaceEnvVar = "WATCH_NAMESPACE"
 
+	// PodNamespaceEnvVar is the env variable for the pod namespace
+	PodNamespaceEnvVar = "POD_NAMESPACE"
```

Comment:
is there any reason we are not using `WatchNamespaceEnvVar`?

Reply:
WatchNamespaceEnvVar is a list of Namespace (from what I recall comma-separated list) the operator should be watching for changes. We don't want the namespace in the onboarding job and provider server to be set as watchNamespace.

We could use OperatorNamespace but this will break if storageCluster is deployed in a different namespace from the operator. Hence using PodNamespace will ensure that these components are deployed in the operator namespace and can get Resources (ex: storageCluster for onboarding job) from the same namespace as it is deployed.